lib/MusicBrainz/Server/Data/URL.pm

```diff
@@ -21,6 +21,7 @@ my %URL_SPECIALIZATIONS = (
     '45cat'               => qr{^https?://(?:www\.)?45cat\.com/}i,
     '45worlds'            => qr{^https?://(?:www\.)?45worlds\.com/}i,
     'Allmusic'            => qr{^https?://(?:www\.)?allmusic\.com/}i,
+    'AmazonMusic'         => qr{^https?://music.amazon(.*?)(?:\:[0-9]+)?/.*/([0-9B][0-9A-Z]{9})(?:[^0-9A-Z]|$)}i,
```

Comment:
This pattern seems strange, since it doesn't care if it matches against something like `music.amazonnotreally.com`, but cares about port numbers which would never appear in the URL (they're not accepted by the URL cleanup in the first place).

(Guess it was mostly copied from the ASIN one below, but judging by 9f9ba4898348b528b2d1e502de3e2319e17a0fb8 that was probably based on a general URL pattern with support for ASINs added at the end.)

It'd seem reasonable to me to just reuse the RegExp added for amazonmusic in the URLCleanup file.